### ISSUE TITLE
Vector dimension restrictions in Grid Function

### ISSUE THREAD

**Issue by vikramsg** [mfem/mfem#162]
I am trying to implement the Euler equations in mfem, and I came across an issue. 

In the Euler equations, the variables are (rho, rho * u, rho * v, E) for 2 space dimensions. So I implemented the variables as a grid function with dimension, dim + 2. 

```
FiniteElementSpace fes(mesh, &fec, dim + 2);
GridFunction u_sol(&fes);
```

I initialized it and found that the last dimension was always taken as 0. So I looked at the code and saw that in fem/fe.cpp we have

```
void NodalFiniteElement::Project (                                                                                                         
   VectorCoefficient &vc, ElementTransformation &Trans, Vector &dofs) const                                                               
{                                                                                                                                             
   MFEM_ASSERT(vc.GetVDim() <= 3, "");                                                                                                             

   double v[3];
```

So the problem is that no vector can be of dimension greater than 3. I do not understand this restriction. This is not the space dimension in which case it would make sense, but why is the vector dimension restricted to 3. 

Can I just remove this restriction? If so does it affect something else that would affect me later on. 

Thanks,
Vikram

**Comment by v-dobrev** [comment]
Hi Vikram,

Yes you can remove the restriction by simply replacing the lines
```c++
   MFEM_ASSERT(vc.GetVDim() <= 3, "");

   double v[3];
   Vector x (v, vc.GetVDim());
```
with the line
```c++
   Vector x(vc.GetVDim());
```
Also, you will need to replace the line
```c++
         dofs(Dof*j+i) = v[j];
```
with
```c++
         dofs(Dof*j+i) = x(j);
```

Hope this helps,
Veselin

**Comment by v-dobrev** [comment]
Fixed in PR #163.

**Comment by vikramsg** [comment]
Thanks Veselin,
